app/js/views/master-view.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'components/react';
+
+import NavigationMenu from 'js/views/navigation-menu';
+
+const DEFAULT_HEADER_CONFIG = Object.freeze({
+  visible: true
+});
+
+const DEFAULT_FOOTER_CONFIG = Object.freeze({
+  visible: true
+});
+
+export default class MasterView extends React.Component {
```

Comment:
Usually this concept is referred to as layout. So maybe rename it to `LayoutView` or just `Layout`.